src/kOS.Safe/Encapsulation/Suffixes/DelegateSuffixResult.cs

```diff
@@ -44,27 +41,24 @@ public void Invoke(ICpu cpu)
             bool isParamArrayArg = false;
 
             CpuUtility.ReverseStackArgs(cpu, false);
-            for (int i = 0; i < paramArray.Length; ++i)
+            for (int i = 0; i < delInfo.Parameters.Length; ++i)
             {
+                DelegateParameter paramInfo = delInfo.Parameters[i];
+
                 object arg = cpu.PopValue();
                 Type argType = arg.GetType();
-                ParameterInfo paramInfo = paramArray[i];
-
-                // If this is the lastmost parameter then it might be a 'param' array which expects all the rest of
-                // the arguments to be collected together into one single array parameter when invoking the method:
-                isParamArrayArg = (i == paramArray.Length - 1 && Attribute.IsDefined(paramInfo, typeof(ParamArrayAttribute)));
```

Comment:
I ran into an issue where kOS threw an `KOSArgumentMismatchException` when executing a delegate suffix.

It doesn't look like there is any point where you are checking to set `isParamArrayArg` to true after removing this line.  I suspect you either intended to check `delInfoParameters` for this information below, or we should add this line back without checking the attribute.  In my quick test, it looks like adding the following line fixes execution:
```
isParamArrayArg = i == delInfo.Parameters.Length - 1 && delInfo.Parameters[i].IsParams;
```

Reply:
ah yeah i accidentally deleted that line 🤐  fixed it when i rebased